src/plugins/intel_gpu/src/runtime/layout.cpp

```diff
@@ -0,0 +1,172 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "intel_gpu/runtime/layout.hpp"
+
+#include <list>
+#include <vector>
+#include <algorithm>
+
+namespace cldnn {
+
+size_t layout::get_rank() const {
+    return format.dimension();
+}
+
+size_t layout::get_spatial_rank() const {
+    return format.spatial_num();
+}
+
+tensor::value_type layout::get_dim(size_t idx) const {
+    auto dims = get_dims();
+    return dims[idx];
+}
+
+tensor::value_type layout::batch() const {
+    auto dims = get_dims();
+    size_t dim_idx = 0;
+    if (!format::is_weights_format(format) && dims[dim_idx] != size.batch[0]) {
+        throw std::runtime_error("batch mismatch: " + std::to_string(dims[dim_idx]) + " vs " + std::to_string(size.batch[0]));
+    }
+    return dims[dim_idx];
+}
+
+tensor::value_type layout::feature() const {
+    auto dims = get_dims();
+    size_t dim_idx = 1;
+    if (!format::is_weights_format(format) && dims[dim_idx] != size.feature[0]) {
+        throw std::runtime_error("feature mismatch: " + std::to_string(dims[dim_idx]) + " vs " + std::to_string(size.feature[0]));
+    }
+    return dims[dim_idx];
+}
+
+tensor::value_type layout::spatial(size_t spatial_idx) const {
+    if (spatial_idx >= format.spatial_num() )
+        return 1;
+    auto dims = get_dims();
+    size_t dim_idx = (format::is_grouped(format) ? 3 : 2) + (format.spatial_num() - 1 - spatial_idx);
+    if (dims[dim_idx] != size.spatial[spatial_idx]) {
+        throw std::runtime_error("spatials mismatch: " + std::to_string(dims[dim_idx]) + " vs " + std::to_string(size.spatial[spatial_idx]));
+    }
+    return dims[dim_idx];
+}
+
+tensor::value_type layout::group() const {
+    auto dims = get_dims();
+    if (!format::is_weights_format(format)) {
+        throw std::logic_error("[GPU] can't get group dimension for data layout");
+    }
+
+    if (!format::is_grouped(format))
+        return 1;
+
+    if (dims[0] != size.group[0]) {
+        throw std::runtime_error("groups mismatch: " + std::to_string(dims[0]) + " vs " + std::to_string(size.group[0]));
+    }
+    return dims[0];
+}
+
+tensor::value_type layout::ofm() const {
+    if (!format::is_weights_format(format)) {
+        throw std::logic_error("[GPU] can't get OFM dimension for data layout");
+    }
+    auto dims = get_dims();
+    size_t dim_idx = format::is_grouped(format) ? 1 : 0;
```

Comment:
It is not related to actual layout, right? I mean, though the layout is ioyx, ofm idx is still 0?